File: cockroachdb/golang/models/responses/response.go
package modelresponses

import "net/http"

type BodyResponse struct {
	Data   interface{} `json:"data"`
	Errors interface{} `json:"errors"`
}

type MessageResponse struct {
	Message string `json:"message"`
}

type Response struct {
	HttpStatusCode int          `json:"httpStatusCode"`
	BodyResponse   BodyResponse `json:"response"`
}

func SetResponse(httpStatusCode int, data interface{}, errors interface{}) Response {
	return Response{
		HttpStatusCode: httpStatusCode,
		BodyResponse: BodyResponse{
			Data:   data,
			Errors: errors,
		},
	}
}

func SetDataResponse(httpStatusCode int, data interface{}) Response {
	return Response{
		HttpStatusCode: httpStatusCode,
		BodyResponse: BodyResponse{
			Data:   data,
			Errors: nil,
		},
	}
}

func SetMessageResponse(message string) Response {
	return Response{
		HttpStatusCode: http.StatusOK,
		BodyResponse: BodyResponse{
			Data: MessageResponse{
				Message: message,
			},
			Errors: nil,
		},
	}
}

func SetOkResponse(data interface{}) Response {
	return Response{
		HttpStatusCode: http.StatusOK,
		BodyResponse: BodyResponse{
			Data:   data,
			Errors: nil,
		},
	}
}

func SetCreateResponse(data interface{}) Response {
	return Response{
		HttpStatusCode: http.StatusOK,
		BodyResponse: BodyResponse{
			Data:   data,
			Errors: nil,
		},
	}
}

func SetNoContentResponse() Response {
	return Response{
		HttpStatusCode: http.StatusNoContent,
		BodyResponse: BodyResponse{
			Data:   nil,
			Errors: nil,
		},
	}
}

func SetBadRequestResponse(message string) Response {
	return Response{
		HttpStatusCode: http.StatusBadRequest,
		BodyResponse: BodyResponse{
			Data: nil,
			Errors: MessageResponse{
				Message: message,
			},
		},
	}
}

func SetInternalServerErrorResponse() Response {
	return Response{
		HttpStatusCode: http.StatusInternalServerError,
		BodyResponse: BodyResponse{
			Data: nil,
			Errors: MessageResponse{
				Message: "internal server error",
			},
		},
	}
}

File: validator/rust/src/models/responses/response.rs
use std::collections::HashMap;
use axum::http::StatusCode;
use serde::Serialize;

#[derive(Debug, Serialize)]
pub struct MessageResponse {
    #[serde(rename = "message")]
    pub message: String
}

#[derive(Debug, Serialize)]
pub struct BodyResponse<T> {
    #[serde(rename = "data")]
    pub data: Option<T>,

    #[serde(rename = "errors")]
    pub errors: Option<HashMap<String, String>>
}

#[derive(Debug)]
pub struct Response<T> {
    pub http_status_code: StatusCode,
    pub body_response: BodyResponse<T>
}

impl <T>Response<T> {
    pub fn set_created_response(data: T) -> Self {
        Self {
            http_status_code: StatusCode::CREATED,
            body_response: BodyResponse {
                data: Some(data),
                errors: None
            }
        }
    }

    pub fn set_bad_request_response(message: String) -> Self {
        let mut error_message = HashMap::new();
        error_message.insert("message".to_string(), message);
        Self {
            http_status_code: StatusCode::BAD_REQUEST,
            body_response: BodyResponse {
                data: None,
                errors: Some(error_message)
            }
        }
    }

    pub fn set_validation_error(validation_error: HashMap<String, String>) -> Self {
        Self {
            http_status_code: StatusCode::BAD_REQUEST,
            body_response: BodyResponse {
                data: None,
                errors: Some(validation_error)
            }
        }
    }

    pub fn set_internal_server_error_response() -> Self {
        let mut error_message = HashMap::new();
        error_message.insert("message".to_string(), "internal server error".to_string());
        Self {
            http_status_code: StatusCode::INTERNAL_SERVER_ERROR,
            body_response: BodyResponse {
                data: None,
                errors: Some(error_message)
            }
        }
    }
}